docs/rules.rst

```diff
@@ -0,0 +1,622 @@
+
+    ``by_role``
+      Optional. Add comment to generated ferm rule definition file that
+      rule is defined in the given Ansible role.
+
+  This template is used among others in the ``debops.libvirtd`` ferm rule
+  `libvirtd__ferm__dependent_rules`_.
+
+* ``default_policy``: Template to define ``iptables`` default policies. The
+  following template-specific YAML keys are supported:
+
+    ``policy``
+      ``iptables`` chain policy, required.
+
+  This template is used in the default rule `ferm__rules_default_policy`_
+  which sets the ``INPUT``, ``FORWARD`` and ``OUTPUT`` chain policies
+  according to `ferm__default_policy_input`_, `ferm__default_policy_forward`_
+  and `ferm__default_policy_output`_.
+
+.. _ferm__rules_default_policy: defaults.html#envvar-ferm__rules_default_policy
+.. _ferm__default_policy_input: defaults.html#envvar-ferm__default_policy_input
+.. _ferm__default_policy_forward: defaults.html#envvar-ferm__default_policy_forward
+.. _ferm__default_policy_output: defaults.html#envvar-ferm__default_policy_output
+
+* ``dmz``: Template to enable connection forwarding to another host. If ``port``
+  is not specified, all traffic is forwarded. The following template-specific
+  YAML keys are supported:
+
+    ``multiport``
+      Optional. Use `iptables multiport`_ extension. Possible values:
+      ``True`` or ``False``. Defaults to ``False``.
+
+    ``public_ip``
+      IPv4 address on the public network which accepts connections, required.
+
+    ``private_ip``
+      IPv4 address of the host on the internal network, required.
+
+    ``protocol(s)``
+      Optional. List of protocols to forward. Defaults to ``tcp``.
+
+    ``port(s)``
+      Optional. List of ports to forward.
+
+    ``dport``
+      Optional. Destination port to forward to. Only needs to be specified
+      if internal destination port is different from the original destination
+      port.
+
+* ``fail2ban``: Template to integrate fail2ban with ``ferm``. As the fail2ban
+  service is defining its own ``iptables`` chains the template will make sure
+  that they are properly refreshed if the ``ferm`` configuration changes.
+
+  This template is used in the default rule `ferm__rules_fail2ban`_.
+
+.. _ferm__rules_fail2ban: defaults.html#envvar-ferm__rules_fail2ban
+
+* ``hashlimit``: Template to define rate limit rules using the
+  `iptables hashlimit`_ extension. The following template-specific YAML
+  keys are supported:
+
+    ``dport``
+      Optional. List of destination ports to which the rule is applied.
+
+    ``enabled``
+      Optional. Enable rule definition. Possible values: ``True`` and
+      ``False``. Defaults to ``True``.
+
+    ``hashlimit_burst``
+      Optional. Number of packets to match within the expiration time.
+      Defaults to ``5``.
+
+    ``hashlimit_expire``
+      Optional. Expiration time of hash entries in seconds. Defaults to
+      ``1.8``.
+
+    ``hashlimit_target``
+      Optional. ? Defaults to ``RETURN``.
```

Comment:
This can be used to specify a chain to jump to when `hashlimit` rules are "tripped" by packets.